examples/motoko_examples/simple-to-do/canisters/rust/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+use std::cell::RefCell;
+use std::collections::BTreeMap;
+use ic_cdk::export::candid::Nat;
+
+#[derive(candid::CandidType, Clone)]
+pub struct ToDo {
+    id: Nat,
+    description: String,
+    completed: bool,
+}
+
+type ToDoList = BTreeMap<Nat, ToDo>;
+
+#[derive(candid::CandidType, Clone)]
+pub struct PerfResult {
+    wasm_body_only: u64,
+    wasm_including_prelude: u64
+}
+
+#[ic_cdk_macros::query]
+pub fn get_perf_result() -> Option<PerfResult> {
+    PERF_RESULT_REF_CELL.with(|perf_result_ref_cell| perf_result_ref_cell.borrow().as_ref().cloned())
+}
+
+thread_local! {
+    static TODO_LIST: RefCell<ToDoList> = RefCell::default();
+    static NEXT_ID: RefCell<Nat> = RefCell::new(Nat::from(0));
+    static PERF_RESULT_REF_CELL: RefCell<Option<PerfResult>> = RefCell::default();
+}
+
+#[ic_cdk_macros::query]
+pub fn get_todos() -> Vec<ToDo> {
+    TODO_LIST.with(|list| {
+        list.borrow().to_owned().into_values().collect()
+    })
+}
+
+// Returns the ID that was given to the ToDo item
+#[ic_cdk_macros::update]
+pub fn add_todo(description: String) -> Nat {
+    let perf_start = ic_cdk::api::call::performance_counter(0);
+
+    let mut id: Nat = Nat::default();
+
+    TODO_LIST.with(|list| {
+        NEXT_ID.with(|next_id| {
+            id = next_id.borrow().clone();
+            list.borrow_mut().insert(id.clone(), ToDo { id: id.clone(), description, completed: false });
+            *next_id.borrow_mut() += 1;
+        })
+    });
+
+    let perf_end = ic_cdk::api::call::performance_counter(0);
+
+    PERF_RESULT_REF_CELL.with(|perf_result_ref_cell| {
```

Comment:
Let's do the same abstraction with `record_performance` that we're doing in Azle and Motoko